web/dataselectie/datasets/brk/views.py

```diff
@@ -269,19 +270,70 @@ def handle_request(self, request, *args, **kwargs):
             raise PermissionDenied("scope BRK/RSN required")
         return super().handle_request(request, *args, **kwargs)
 
-    def filter_data(self, elastic_data, request):
+    def load_from_elastic(self) -> dict:
         """
-        Remove duplicate kadastraal_object_id from object_list
+        Overwrite method of TableSearchView.load_from_elastic.
+        This overwrite make the result of kadastraal ID's unique
+        Due to that multiple addresses can be related to one kadastraal ID
+        that causes the kadastraal ID to appear more than once.
         """
-        new_object_list = []
-        kot_unique = set()
-        for object in elastic_data['object_list']:
-            kadastraal_object_id = object['kadastraal_object_id']
-            if kadastraal_object_id not in kot_unique:
-                new_object_list.append(object)
-                kot_unique.add(kadastraal_object_id)
-
-        elastic_data['object_list'] = new_object_list
+        # looking for a query
+        query_string = self.request_parameters.get('query', None)
+
+        # Building the query
+        q = self.elastic_query(query_string)
+        query = self.add_elastic_filters(q)
+
+        # remove size and page filter
+        # because after manipulating the result for unique kadastraal IDs
+        # the returned rows must be restricted by the size and page paramaters
+        # not before.
+        param_size = query['size']
+        try:
+            param_page = int(self.request_parameters.get('page', 1))
+        except ValueError:
+            param_page = 1
+        offset = (param_page - 1) * param_size
+        size = param_size * param_page if offset > 0 else param_size
+
+        # reset orginal param values (the param_size and offset are used now)
+        query['size'] = settings.MAX_SEARCH_ITEMS
+        query['from'] = 0
+
+        # Performing the search
+        response = self.elastic.search(
+            index=settings.ELASTIC_INDICES[self.index], body=query)
+
+        # identify unique kadastraal objects
+        kadastraal_unique_set = set()
```

Comment:
You seem to be filtering in Python code. That's not efficient. Not really familiar with ElasticSearch, but a bit of DuckDuckGo'ing it seems ElasticSearch can do this for you: https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html

Reply:
Wow, die had ik gemist. Werkt prima :+1: